IMHO it is not useful to check for "empty password" when authenticating
with pubkey, so disable this check.
It is still not allowed to log in with empty password and passwd auth.

diff --git a/svr-auth.c b/svr-auth.c
index 5da0aa7..aadcbcc 100644
--- a/svr-auth.c
+++ b/svr-auth.c
@@ -248,6 +248,7 @@ static int checkusername(unsigned char *username, unsigned int userlen) {
 		return DROPBEAR_FAILURE;
 	}
 
+#if 0
 	/* check for an empty password */
 	if (ses.authstate.pw_passwd[0] == '\0') {
 		TRACE(("leave checkusername: empty pword"))
@@ -256,6 +257,7 @@ static int checkusername(unsigned char *username, unsigned int userlen) {
 		send_msg_userauth_failure(0, 1);
 		return DROPBEAR_FAILURE;
 	}
+#endif
 
 	TRACE(("shell is %s", ses.authstate.pw_shell))
 
